trace/trace.go

```diff
@@ -321,6 +326,12 @@ func (s *Span) send() {
 		}
```

Comment:
it looks like there are a few places here and in the `Send()` method where we call `s.ev.AddField` instead of `s.AddField`. This is fine currently, but it might be worth converting to `s.AddField` throughout to avoid potentially introducing similar race conditions in the future?